pkg/backend/identityapi/api_client.go

```diff
@@ -0,0 +1,23 @@
+package identityapi
+
+import (
+	"context"
+	"github.com/newrelic/infrastructure-agent/pkg/identity-client"
+	"net/http"
+)
+
+type apiClient interface {
+	RegisterPost(
+		ctx context.Context,
+		userAgent string,
+		xLicenseKey string,
+		registerRequest identity.RegisterRequest,
+		localVarOptionals *identity.RegisterPostOpts) (identity.RegisterResponse, *http.Response, error)
+
+	RegisterBatchPost(
+		ctx context.Context,
+		userAgent string,
+		xLicenseKey string,
+		registerRequest []identity.RegisterRequest,
+		localVarOptionals *identity.RegisterBatchPostOpts) ([]identity.RegisterBatchEntityResponse, *http.Response, error)
```

Comment:
non batch returns: `identity.RegisterResponse`
while batch returns: `identity.RegisterBatchEntityResponse`

and `identityapi` has:
- `RegisterBatchEntityResponse`
- `RegisterEntityResponse`

1. can we replace `identity.RegisterBatchEntityResponse` for consistency with its counterpart and to avoid confusion?
2. are both package types required? couldn't we get rid of a couple of these packages types?